workflow-templates/shellcheck.yml

```diff
@@ -0,0 +1,65 @@
+# ShellCheck is a linter for your shell scripts:
```

Comment:
discussion from https://github.com/openedx/.github/pull/60 (@feanil): 

![image](https://user-images.githubusercontent.com/3628148/221641987-cd33f9ec-12ee-4127-a1cf-97e450ebe82d.png)

Reply:
@feanil @timmc-edx I'm strongly leaning towards dropping the dependency on the `ludeeus/action-shellcheck` action, and instead rolling our own simple shared shellcheck workflow. I am imagining that the shared workflow would accept arguments:

* shellcheck version
* root directory (defaulting to `.`)
* file ignore patterns (optional)
* shellcheck options (optional)

and it would just need to:

1. Install shellcheck at the specified version.
2. Recursively list files, starting from the root directory, and applying any supplied file ignore patterns.
3. Call shellcheck on the list of files, fowarding any supplied shellcheck options.

Steps 2-3 would be implemented as one-line shell command so that users could copy-paste it into their local terminal and see the same output.

Do either of you have any objections before I work on implementing that?

Reply:
I think it's fine to roll our own steps but I would still wrap it in a shared workflow to ease making updates and changing underlying implementation easier.

Reply:
Agreed 100%. I don't want to put any non-trivial logic in the workflow templates, so the shared workflow would be a must.

Reply:
This is done.